diff --git a/mealie/core/settings/settings.py b/mealie/core/settings/settings.py
index 7558a29c..e77a0506 100644
--- a/mealie/core/settings/settings.py
+++ b/mealie/core/settings/settings.py
@@ -69,7 +69,7 @@ def get_secrets_dir() -> str | None:
 
     logger = get_logger()
 
-    secrets_dir = "/run/secrets"
+    secrets_dir = os.environ.get("SECRETS_DIR", "/run/secrets")
 
     # Check that the secrets directory exists.
     if not os.path.exists(secrets_dir):
